Userland/Libraries/LibAudio/WavLoader.cpp

```diff
@@ -89,58 +84,40 @@ void WavLoaderPlugin::seek(const int position)
         m_stream->seek(byte_position);
 }
 
-void WavLoaderPlugin::reset()
-{
-    seek(0);
-}
-
 bool WavLoaderPlugin::parse_header()
 {
-    OwnPtr<Core::IODeviceStreamReader> file_stream;
-    bool ok = true;
-
+    OwnPtr<Core::InputFileStream> file_stream;
     if (m_file)
-        file_stream = make<Core::IODeviceStreamReader>(*m_file);
+        file_stream = make<Core::InputFileStream>(*m_file);
+
+    AK::InputStream* const stream =
+        (m_file ?
+            file_stream.ptr() :
+            dynamic_cast<AK::InputStream*>(m_stream.ptr()));
```

Comment:
It feels awkward to do this dance here.

Could we refactor the class further to have `m_stream` be an `OwnPtr<InputStream>` and assign to it `make<InputMemoryStream>(buffer)` in the ByteBuffer constructor and `make<InputFileStream>(m_file)` in the const StringView& constructor? Not sure how well OwnPtr works with OwnPtr<Base> = OwnPtr<Derived> though.. that's something std:::unique_ptr handles well.

This transformation would allow removing all of the `if m_file --> do things to m_file;  else --> do things to m_stream` and always use m_stream.

Reply:
One snag - `AK::InputStream` does not define `seek`. I don't think it makes conceptual sense to add `AK::InputStream::seek`, and it would also affect a lot of other classes.

So I think I'll still need one instance of `if (m_file)/else`.